src/zenml/models/v2/core/artifact_version.py

```diff
@@ -116,6 +128,14 @@ class ArtifactVersionResponseBody(WorkspaceScopedResponseBody):
         title="URI of the artifact.", max_length=STR_FIELD_MAX_LENGTH
     )
     type: ArtifactType = Field(title="Type of the artifact.")
+    materializer: Source = Field(
+        title="Materializer class to use for this artifact.",
+    )
+    data_type: Source = Field(
+        title="Data type of the artifact.",
+    )
+
+    _convert_source = convert_source_validator("materializer", "data_type")
```

Comment:
Yep, we need those properties to load an artifact and having them in the body saves us one request per step input artifact. It can't be solved using hydration alone as these are part of the `StepRunResponse` and we would require a second level of hydration if we wanted to access the `ArtifactResponseMetadata`.